diagnostic_rt_qpcr/libraries/diagnosticrtqpcrcompositions/source.rb

```diff
@@ -15,25 +17,64 @@ module DiagnosticRTqPCRCompositions
   #
   # @param operations [OperationList]
   # @return [void]
-  def build_stripwell_master_mix_compositions(operations:)
+  def build_stripwell_primer_probe_compositions(operations:)
+    all_inputs = []
     operations.each do |op|
-      primer_mixes = []
-      op.input_array(PRIMER_PROBE_MIX).each do |fv|
-        primer_mixes += fv.collection.parts
+      options = op.temporary[:options]
+      program_name = options[:program_name]
+
+      stripwells = find_stripwells(sample_names: options[:sample_names],
+                                   stripwell_ot: options[:object_type],
+                                   dimensions: op.temporary[:microtiter_plate].collection.dimensions,
+                                   all_inputs: all_inputs)
+      if stripwells.any?(&:empty?)
+        op.error(ProtocolError, 'No enough parts in inventory')
+        next
       end
 
-      compositions = []
-
-      primer_mixes.each do |primer_mix|
-        composition = build_modified_master_mix_composition(
-          primer_mix: primer_mix,
-          program_name: op.temporary[:options][:program_name]
-        )
-        compositions.append(composition)
-      end
+      parts_by_row = convert_to_parts_by_row(stripwells: stripwells)
 
+      compositions = setup_composition_parts(parts_by_row: parts_by_row,
+                                             program_name: program_name)
       op.temporary[:compositions] = compositions
+      all_inputs.push(stripwells.flatten)
     end
+    all_inputs.flatten
+  end
+  
+  # Converts to parts by row format.  Mostly to allow for easy conversion
+  # if the primer probes ever start coming in as individual parts and
+  # not stripwells
+  def convert_to_parts_by_row(stripwells: stripwells)
+    stripwells.map(&:parts)
+  end
+  
+  def setup_composition_parts(parts_by_row:, program_name:)
+    compositions = []
+    parts_by_row.each do |parts|
+      parts.each do |item|
+        compositions.push(build_primer_probe_composition(item: item,
+                                        program_name: program_name))
+      end
+    end
+    compositions
+  end
+  
+  def find_stripwells(sample_names:, stripwell_ot:, dimensions:, all_inputs:)
+    ordered_sample_names = Array.new(dimensions[0]/(sample_names.length), sample_names).flatten
+    stripwells = []
+    ordered_sample_names.each do |name|
+      possible_items = Item.where(sample: Sample.find_by_name(name))
+                           .to_ary
+                           .select(&:is_part)
+                           .map(&:containing_collection)
+                           .uniq
+                           .select { |col| col.object_type.name == stripwell_ot }
+                           .map(&:id)
+      unused_items = (possible_items - stripwells.map(&:id)) - all_inputs.map(&:id)
```

Comment:
`stripwells.map(&:id)` doesn't seem like it should work because it's an array of arrays. A more elegant way to approach this, and to avoid passing the `all_inputs` object might be to change the location of the stripwell to 'provisioned' once it is spoken for. 

Incidentally, I don't think this method safely excludes deleted items.